modules/fedimint-ln-client/src/lib.rs

```diff
@@ -699,16 +731,74 @@ impl LightningClientModule {
         }
     }
 
+    async fn register_payment_hash_with_gateway(
+        &self,
+        gateway: &LightningGateway,
+        payment_hash: sha256::Hash,
+        amount: Amount,
+        description: String,
+        expiry_time: Duration,
+    ) -> anyhow::Result<Bolt11Invoice> {
+        let response = reqwest::Client::new()
+            .post(
+                gateway
+                    .api
+                    .join("register_payment_hash")
+                    .expect("register_payment_hash contains no invalid characters for a URL")
+                    .as_str(),
+            )
+            .json(&RegisterPaymentHashPayload {
+                amount,
+                description: description.clone(),
+                expiry_time,
+                payment_hash,
+                federation_id: self.client_ctx.get_config().global.federation_id(),
+            })
+            .send()
+            .await
+            .context("Failed to register payment hash with gateway")?;
+        if !response.status().is_success() {
+            bail!(
+                "Failed to register payment hash with gateway. Returned error code: {}",
+                response.status()
+            );
+        }
+
+        let invoice: Bolt11Invoice = response.json().await?;
```

Comment:
probably makes sense to have this as a json object so it can be more extensible in the future